### PR TITLE
SmartXSP Bid Adapter : initial release

### PR DESCRIPTION
**Pull request by olivierschaal**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [X] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

We propose the first version of our Prebid.js bidder
<!-- For new bidder adapters, please provide the following
- contact email of the adapter’s maintainer
pbjs@smartxsp.io
- test parameters for validating bids:
```
{
  bidder: 'smartxsp',
  params: {
    placementId: 13144370,
    token: '1234'
  }
}
```

### CODE REVIEW

**Comment by olivierschaal** [comment]
Hello does this first version suit requirements?

**Comment by ChrisHuie** [comment]
@olivierschaal it looks like our integration tests didn't run for this pr. This is usuallly because you are following your fork on CircleCI and not the main Prebid projets. You can see the projects you are following on the left hand side of CircleCII

File: modules/smartxspBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        bidObject = {
+          requestId: bid.bidId,
+          cpm: ad.cpm,
+          width: parseInt(ad.bannerFormatWidth),
+          height: parseInt(ad.bannerFormatHeight),
+          creativeId: ad.id,
+          netRevenue: !!ad.netRevenue,
+          currency: ad.currency,
+          ttl: ad.ttl,
+          ad: ad.ad
```
Comment (ChrisHuie) [inline modules/smartxspBidAdapter.js:105]: Can you please add support for adomain and schain as well

**Comment by ChrisHuie** [comment]
Please also submit docs to our docs repo -> https://github.com/prebid/prebid.github.io/pulls

**Comment by olivierschaal** [comment]
Hello, we create the pull request for the doc and add the management of adomain and schain in the bidder.

**Comment by ChrisHuie** [comment]
@olivierschaal this lgtm. Can you just pull in master to run our CircleCi implementation tests on this pr 🙏

**Comment by olivierschaal** [comment]
> @olivierschaal this lgtm. Can you just pull in master to run our CircleCi implementation tests on this pr 🙏

Sorry Chris but I don't understand what I have to do ?

**Comment by ChrisHuie** [comment]
@olivierschaal can you just pull in recent commits from master into this pr? Just want to double check that it syncs with changes since this was submitted 🙏

**Comment by ChrisHuie** [comment]
Sorry. Will merge this today 👍

**Comment by ChrisHuie** [comment]
@olivierschaal tried to merge this but there are some linting issues popping up from testing

 67:9  error  Identifier 'bid_url' is not in camel case  camelcase
  69:7  error  Identifier 'bid_url' is not in camel case  camelcase
  81:5  error  Unexpected console statement               no-console
  85:9  error  'url' is defined but never used            no-unused-vars
  86:9  error  'html' is defined but never used           no-unused-vars
  89:9  error  'accountId' is defined but never used      no-unused-vars
  90:9  error  'language' is defined but never used       no-unused-vars
  91:9  error  'scriptId' is defined but never used       no-unused-vars

File: modules/smartxspBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          'banner-format-height': sizes[j][1]
+        }
+        if (bid.params.bannerFormat) {
+          bidParam['banner-format'] = bid.params.bannerFormat
+        }
+        if (bid.params.language) {
+          bidParam.language = bid.params.language
+        }
+        if (bid.params.region) {
+          bidParam.region = bid.params.region
```
Comment (patmmccann) [inline modules/smartxspBidAdapter.js:52]: not covered in tests, how is this used

File: modules/smartxspBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          accountId: bid.params.accountId || '',
+          bidId: bid.bidId,
+          'banner-format-width': sizes[j][0],
+          'banner-format-height': sizes[j][1]
+        }
+        if (bid.params.bannerFormat) {
+          bidParam['banner-format'] = bid.params.bannerFormat
+        }
+        if (bid.params.language) {
+          bidParam.language = bid.params.language
```
Comment (patmmccann) [inline modules/smartxspBidAdapter.js:49]: read from standard locations

File: test/spec/modules/smartxspBidAdapter_spec.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    let request = spec.buildRequests(bidRequests, bidderRequests);
+    it('sends bid request via POST', function () {
+      expect(request.method).to.equal('POST');
+    });
+    it('must contain accountId', function() {
+      expect(request.data.bidParams[0].accountId).to.equal('29Md1Mx1x2MpM7Me');
+    });
+
+    it('must contain auctionId', function() {
+      expect(request.data.auctionId).to.exist.and.to.be.a('string')
```
Comment (patmmccann) [inline test/spec/modules/smartxspBidAdapter_spec.js:54]: this will often be null in prebid 8

**Review by patmmccann — CHANGES_REQUESTED** [review]

**Comment by olivierschaal** [comment]
Hi @ChrisHuie is everything correct ?

File: test/spec/modules/smartxspBidAdapter_spec.js (hunk 4 above)
Comment (patmmccann) [reply test/spec/modules/smartxspBidAdapter_spec.js:56]: you marked this as resolved but the title of your test suggest you cannot take nulls here. If you are indeed able to, perhaps rename the the test as 'auctionid passed when available'

File: modules/smartxspBidAdapter.js (hunk 3 above)
Comment (patmmccann) [reply modules/smartxspBidAdapter.js:46]: this is not resolved, you should read language from the openrtb location eg how yahoo does here https://github.com/prebid/Prebid.js/blob/509deaa0649824369907e6b76d0a4d045ec84df8/modules/yahoosspBidAdapter.js#L442

**Review by patmmccann — CHANGES_REQUESTED** [review]
couple things remain unaddressed

**Comment by patmmccann** [comment]
Closing for lack of engagement

**Pull request closed without merging**